pymatgen/io/core.py

```diff
@@ -0,0 +1,240 @@
+"""
+This module defines the abstract interface for pymatgen InputSet and InputSetGenerator classes.
+"""
+
+import abc
+import os
+from pathlib import Path
+from collections.abc import Mapping
+from string import Template
+from typing import Union, Optional, Dict
+from zipfile import ZipFile
+from monty.json import MSONable
+from monty.io import zopen
+
+
+__author__ = "Ryan Kingsbury"
+__email__ = "RKingsbury@lbl.gov"
+__status__ = "Development"
+__date__ = "September 2021"
+
+
+class InputFile(MSONable):
+    """
+    Abstract base class to represent a single input file. Note that use
+    of this class is optional; it is possible create an InputSet that
+    does not rely on underlying Inputfile objects.
+
+    All InputFile classes must implement a __str__ method, which
+    is called by write_file.
+    """
+
+    @abc.abstractmethod
+    def __str__(self) -> str:
+        """
+        String representation of an entire input file.
+        """
+
+    def write_file(self, filename: Union[str, Path]) -> None:
```

Comment:
shorten to write.

Reply:
Fine. But just a note that backwards compatibility is not an issue. You can always have both write and write_file with a deprecation message for write_file.